### DISCUSSION TITLE
Unsteady laminar incompressible flow around a cylidner Re=200

### DISCUSSION THREAD

**Discussion by suargi** [su2code/SU2#1958] [Q&A]
Hi,

I would like to solve the incompressible laminar NS around a 2d cylinder at Reynolds 200, which should produce an unsteady solution. I am having some problems getting mesh converged results - I am monitoring the drag coefficient. One possibility might be due to the fact that the drag value does not fully converge within each time iteration. Analysing the residuals, I can see that within each time step:
 - Velocity: it converges 3 orders of magnitude.
 - Pressure: I am lucky if it converges one order of magnitude.

At the beginning the initial pressure residual is in the order of magnitude of -3. After 1000 time steps, the initial pressure residual is around -8. However, the behaviour explained above can be seen all along the simulation, no matter the initial residual value.

I have included the config file and one of the meshes. Besides, I have enclosed the output of SU2 which gathers the convergence of the residuals. You can download them from [here](https://seafile.rlp.net/f/e36375528d21474bb2e4/?dl=1).

Any idea on how to make the pressure converge?

Thank you!

**Comment by pcarruscag** [comment]
There are some pressure waves in the domain, I messed with several settings and terms in the code and ended up noticing that with MUSCL off things worked ok, since it was only the pressure that had weird waves I disabled only the pressure reconstruction and still used MUSCL for all other quantities (using limiters did not help). The results also look ok (convergence-wise and visually, I didn't take any numbers).
Maybe our boundary treatment introduces too many reflections and we need the extra dissipation to cope (pressure is sensitive in incompressible flows because d rho / d t = 0), or maybe there is some little bug in the FDS dissipation term.

After writing the above I thought of trying some artificial compressibility in the dual time term, this also works ok even with MUSCL reconstruction of the pressure. So maybe it's just the boundaries, but I don't know what to do better there.

**Reply by suargi** [reply]
Thank you for the detailed study. Setting MUSCL off for all the variables the solution reaches a steady state regime, meanwhile with MUSCL on I can get the expected vortex shedding.
I would like to give a try setting MUSCL off only for the pressure, as you did. Could you please give some insights?

**Reply by pcarruscag** [reply]
In CIncEulerSolver::Upwind_Residual, inside the if (muscl) branch, after reconstructing all variables, I added:
Primitive_i[0] = V_i[0];
Primitive_j[0] = V_j[0];
I also had the SetPreconditioner modification to BetaInc2 in my tests.
